commons/src/main/java/com/powsybl/commons/config/MapModuleConfig.java

```diff
@@ -64,41 +63,70 @@ private static String substitureEnvVar(String str) {
         return str.replace("$HOME", System.getProperty("user.home"));
     }
 
+    private static PowsyblException createPropertyNotSetException(String name) {
+        return new PowsyblException("Property " + name + " is not set");
+    }
+
+    private static PowsyblException createUnexpectedPropertyTypeException(Object value) {
+        return new PowsyblException("Unexpected property value type: " + value.getClass().getName());
+    }
+
+    public Optional<String> getOptionalStringProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return Optional.empty();
+        }
+        if (!(value instanceof String)) {
+            throw createUnexpectedPropertyTypeException(value);
+        }
+        return Optional.of((String) value).map(MapModuleConfig::substitureEnvVar);
+    }
+
     @Override
     public String getStringProperty(String name) {
-        String value = (String) properties.get(name);
-        if (value == null || value.trim().isEmpty()) {
-            throw new PowsyblException("Property " + name + " is not set");
-        }
-        return substitureEnvVar(value);
+        return getOptionalStringProperty(name).orElseThrow(() -> createPropertyNotSetException(name));
     }
 
     @Override
     public String getStringProperty(String name, String defaultValue) {
-        Object value = properties.get(name);
-        return substitureEnvVar(value == null ? defaultValue : (String) value);
+        return getOptionalStringProperty(name).orElse(defaultValue);
     }
 
     public void setStringProperty(String name, String value) {
         properties.put(Objects.requireNonNull(name), Objects.requireNonNull(value));
     }
 
+    public Optional<List<String>> getOptionalStringListProperty(String name) {
+        Objects.requireNonNull(name);
+        Object value = properties.get(name);
+        if (value == null) {
+            return Optional.empty();
+        } else {
+            if (value instanceof String) {
+                return Optional.of(new ArrayList<>(Arrays.asList((substitureEnvVar((String) value)).split("[:,]"))));
```

Comment:
Is it necessary to convert the immutable list to a mutable one?

Reply:
Not sure, I removed, let's see